src/utils/unbanScheduler.ts

```diff
@@ -0,0 +1,58 @@
+import { Client } from "discord.js";
+import { removeModeration, getPendingBans } from "./database/moderation";
+
+const scheduledUnbans = new Map<string, NodeJS.Timeout>();
+
+export function scheduleUnban(client: Client, guildId: string, userId: string, delay: number) {
+  const key = `${guildId}-${userId}`;
+
+  if (scheduledUnbans.has(key)) {
+    clearTimeout(scheduledUnbans.get(key)!);
+  }
+
+  // 24.8 days because why not
+  const maxDelay = 2147483647; 
+
+  if (delay > maxDelay) {
+    const remainingDelay = delay - maxDelay;
+    const timeout = setTimeout(() => {
+      scheduleUnban(client, guildId, userId, remainingDelay);
+    }, maxDelay);
```

Comment:
What happens if it goes negative? I never tested that, will it count as a timer of 0 ?

Reply:
I will check later. And if unbans instantly or something I will add a fix